### ISSUE TITLE
Sensor group weather/sensor data is applied to one zone then deleted when other zone depend on the data 

### ISSUE THREAD

**Issue by jata1** [jeroenterheerdt/HAsmartirrigation#325]
### What happened?

I have 2 zones setup with one sensor group. When I manually update/calculate a zone the sensor data is deleted and is not available for other zones that depend on the same sensor data.

### How to reproduce

1. create 2 zones (zone 1 and zone 2) and link both to the same sensor group.
2. wait until sensor group data is collected then apply/calculate for zone 1 using the zone specific buttons in the GUI 
3. note now that the data is deleted and is not available to calculate watering for zone 2

### Which version are you running?

v2023.12.2

### Diagnostics file

- [X] I have attached a diagnostics file

### Additional information

Not sure what happens if I click the 'calculate all zones' button instead of the calculation for each zone separately.

diagnostic file attached...

[config_entry-smart_irrigation-7f7127f4c22a04d1cbd475295c54538c.json.txt](https://github.com/jeroenterheerdt/HAsmartirrigation/files/14017564/config_entry-smart_irrigation-7f7127f4c22a04d1cbd475295c54538c.json.txt)

**Comment by jata1** [comment]
as requested @jeroenterheerdt. 

To be clear what i mean by apply/calculate for a zone, see screen shot below.

![image](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/23280705/6208e1b5-189a-4e11-99bf-022e0e7fa2f1)

**Comment by jeroenterheerdt** [comment]
ah, that is by design. If you use the 'calculate all' button or service it should work, as in: we make sure to calculate everything only once / use the same weatherdata many times over  If you press the button you highlighted then yes, the weather data is deleted.

**Comment by jata1** [comment]
I need to check what my automations are doing as I could have something that calculates for a specific zone and I assume this will also delete the data

**Comment by jeroenterheerdt** [comment]
> haha. That explains why I have been going mad for the last week when one zone calculated fine and the other did nothing.
> 
> I think you need to add a note to explain this (or a warning) to say to use the calculate all zones

that's fair. The UI is just for debugging. We expect users to just use the calculate all service.

**Comment by jeroenterheerdt** [comment]
> I need to check what my automations are doing as I could have something that calculates for a specific zone and I assume this will also delete the data

correct.

**Comment by jata1** [comment]
> > haha. That explains why I have been going mad for the last week when one zone calculated fine and the other did nothing.
> > I think you need to add a note to explain this (or a warning) to say to use the calculate all zones
> 
> that's fair. The UI is just for debugging. We expect users to just use the calculate all service.

well I guess I am an unexpected user lol

![image](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/23280705/dca4b4ba-5155-40eb-af95-566bccce47d7)

**Comment by jata1** [comment]
the above is fired when my tap/value stops watering. This was also not working properly until I created a duplicate sensor group.

I wanted this to work like this so I get an estimate of how much more watering a zone 'needs' when the valve turns off. This is so I can manually water and still know how much water is needed

**Comment by jeroenterheerdt** [comment]
> > haha. That explains why I have been going mad for the last week when one zone calculated fine and the other did nothing.
> > I think you need to add a note to explain this (or a warning) to say to use the calculate all zones
> 
> that's fair. The UI is just for debugging. We expect users to just use the calculate all service.

I have updated the readme to clarify it.

**Comment by jeroenterheerdt** [comment]
> the above is fired when my tap/value stops watering. This was also not working properly until I created a duplicate sensor group.
> 
> I wanted this to work like this so I get an estimate of how much more watering a zone 'needs' when the valve turns off. This is so I can manually water and still know how much water is needed

you can then just calculate all zones, right? you'd still have the data you need to manually water?

**Comment by jata1** [comment]
> > the above is fired when my tap/value stops watering. This was also not working properly until I created a duplicate sensor group.
> > I wanted this to work like this so I get an estimate of how much more watering a zone 'needs' when the valve turns off. This is so I can manually water and still know how much water is needed
> 
> you can then just calculate all zones, right? you'd still have the data you need to manually water?

yep. just need to change the two automations/scripts to calculate all zones.

**Comment by jata1** [comment]
Overall I think it would be cleaner/better to be able to do the things I wanted for each zone but I get it will be tricky as you'll need to hold sets of calculated data for each zone that is using a sensor group.

Good that you have clarified in the readme / documentation

**Comment by jeroenterheerdt** [comment]
Yeah, I don't want to keep copies of the data around and it can grow really large very quickly. I mean, I could potentially create a parameter to the service to control deletion after calculating that defaults to true. That would not break existing users but provide flexibility for you, right?
